cmd/registry/core/visitors.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package core
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/apigee/registry/pkg/connection"
+	"github.com/apigee/registry/server/registry/names"
+)
+
+type Visitor interface {
+	ProjectHandler() ProjectHandler
+	ApiHandler() ApiHandler
+	DeploymentHandler() DeploymentHandler
+	DeploymentRevisionHandler() DeploymentHandler
+	VersionHandler() VersionHandler
+	SpecHandler() SpecHandler
+	SpecRevisionHandler() SpecHandler
+	ArtifactHandler() ArtifactHandler
+}
+
+type VisitorOptions struct {
+	RegistryClient   connection.RegistryClient
+	AdminClient      connection.AdminClient
+	Pattern          string
+	Filter           string
+	AllowUnavailable bool
```

Comment:
`AllowUnavailable` is specific to Project... should we name it `AllowUnavailableProject` to make it more clear what this actually does?

Reply:
Yes, sounds good, I'll make that change.